pkg/daemon/daemon.go

```diff
@@ -981,19 +949,25 @@ func (p *ptpProcess) ProcessTs2PhcEvents(ptpOffset float64, source string, iface
 	}
 }
 
-func (dn *Daemon) getHaProfiles(profiles string) []string {
-	haProfiles := strings.Split(profiles, ",")
-	updateHaProfileToSocketPath := make([]string, 0, len(haProfiles)-1)
-	for _, profileName := range haProfiles {
-		if strings.TrimSpace(profileName) != "" {
-		next:
-			for _, dmProcess := range dn.processManager.process {
-				if strings.TrimSpace(*dmProcess.nodeProfile.Name) == strings.TrimSpace(profileName) {
-					updateHaProfileToSocketPath = append(updateHaProfileToSocketPath, "-z "+dmProcess.ptp4lSocketPath)
-					continue next
+func (dn *Daemon) applyHaProfiles(nodeProfile *ptpv1.PtpProfile, cmdLine string) string {
+	if profiles, ok := (*nodeProfile).PtpSettings[PTP_HA_IDENTIFIER]; ok {
+		haProfiles := strings.Split(profiles, ",")
+		updateHaProfileToSocketPath := make([]string, 0, len(haProfiles)-1)
+		for _, profileName := range haProfiles {
+			if strings.TrimSpace(profileName) != "" {
+			next:
+				for _, dmProcess := range dn.processManager.process {
+					if strings.TrimSpace(*dmProcess.nodeProfile.Name) == strings.TrimSpace(profileName) {
+						updateHaProfileToSocketPath = append(updateHaProfileToSocketPath, "-z "+dmProcess.ptp4lSocketPath)
+						continue next
```

Comment:
I don't understand why the 'next' is needed. Wouldn't this lead to an infinite loop?